platform/platform-logging/src/main/java/org/codice/ddf/logging/platform/LoggingServiceBean.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Override
+    public List<LogEvent> retrieveLogEvents() {
+        try {
+            readLock.lock();
+            return Lists.newArrayList(logEvents);
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    /**
+     * Sets the maximum number of {@link LogEvent}s to store
+     * 
+     * @param newMaxLogEvents
+     *            This number cannot be less than 0 or greater than {@code MAX_LOG_EVENTS_LIMIT}. In
+     *            the event that this parameter is less than 0 or greater than
+     *            {@code MAX_LOG_EVENTS_LIMIT}, the maximum log events stored will set set to
+     *            {@code MAX_LOG_EVENTS_LIMIT}.
+     */
+    public void setMaxLogEvents(int newMaxLogEvents) {
+        try {
+            writeLock.lock();
+            if (newMaxLogEvents <= 0 || newMaxLogEvents > MAX_LOG_EVENTS_LIMIT) {
+                LOGGER.warn(
+                        "An invalid value of [{}] was entered for maximum log events to store. This "
+                                + "value must be greater than 0 and must not exceed [{}]. Maximum log"
+                                + " events to store will be set to [{}].", newMaxLogEvents,
+                        MAX_LOG_EVENTS_LIMIT, DEFAULT_LOG_EVENTS_LIMIT);
+                newMaxLogEvents = DEFAULT_LOG_EVENTS_LIMIT;
+            }
+
+            EvictingQueue<LogEvent> evictingQueue = EvictingQueue.create(newMaxLogEvents);
+
+            if (logEvents.size() < newMaxLogEvents) {
+                evictingQueue.addAll(logEvents);
+            } else {
+                Iterable<LogEvent> iterable = Iterables.skip(logEvents, logEvents.size()
+                        - newMaxLogEvents);
+                evictingQueue.addAll(Lists.newArrayList(iterable));
+            }
+            this.maxLogEvents = newMaxLogEvents;
+            logEvents = evictingQueue;
+        } finally {
+            writeLock.unlock();
+        }
+    }
+
+    public int getMaxLogEvents() {
+        try {
+            readLock.lock();
+            return maxLogEvents;
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    private void add(LogEvent logEvent) {
+        try {
+            writeLock.lock();
+            logEvents.add(logEvent);
+        } finally {
+            writeLock.unlock();
+        }
+    }
+
+    private LogEvent createLogEvent(PaxLoggingEvent paxLoggingEvent) {
+        long timestamp = paxLoggingEvent.getTimeStamp();
+        String level = paxLoggingEvent.getLevel().toString();
+        String message = paxLoggingEvent.getMessage();
+        String bundleName = getBundleName(paxLoggingEvent);
+        String bundleVersion = getBundleVersion(paxLoggingEvent);
+        LogEvent logEvent = new LogEvent(timestamp, level, message, bundleName, bundleVersion);
+        return logEvent;
+    }
+
+    private String getBundleName(PaxLoggingEvent paxLoggingEvent) {
+        return (String) paxLoggingEvent.getProperties().get(BUNDLE_NAME);
```

Comment:
Since `BUNDLE_NAME` isn't the _actual_ name of the bundle, but a value used to retrieve it, maybe we rename it to `BUNDLE_NAME_KEY` for clarity?